src/CodegenFile.php

```diff
@@ -330,6 +339,30 @@ public function setGeneratedFrom(
     return $this;
   }
 
+  public function setNamespace(string $file_namespace): this {
+    $this->fileNamespace = $file_namespace;
+    return $this;
+  }
+
+  public function useNamespace(string $ns, ?string $as = null): this {
+    $this->useNamespaces[] = $as === null
+      ? "use $ns;"
+      : "use $ns as $as;";
```

Comment:
Can we just store it as a vector of pairs and only transform to string when writing it to code ? That'd allow us to do basic checks like using namespace twice